components/ide/code/codehelper/main.go

```diff
@@ -61,14 +61,21 @@ func main() {
 	}
 	phaseDone()
 
+	url, err := url.Parse(wsInfo.GitpodHost)
```

Comment:
We have only codehelper which should work with all images. Let's keep backward compatible mode based on version from product.json.

We will need to keep old rules in installer as well.

We can only remove them when all Dedicated cells and Cloud are migrated to new images.

Reply:
This is just :lipstick: as `GitpodHost` was parsed twice in `prepareServerMain` and `prepareWebWorkbenchMain`, the functions will still be invoked correctly for all versions, `GITPOD_CODE_HOST` env var will be ignored for old versions